src/themes/index.story.js

```diff
@@ -0,0 +1,28 @@
+export function getCssColorVars() {
```

Comment:
I find this implementation clearer
```js
export function getCssColorVars() {
  const styleSheet = [...document.styleSheets].find((sheet) => sheet.href?.includes('web-components-library'));

  const themeVariables = [...styleSheet.cssRules]
    .filter((rule) =>
      // only take the declarations for :root and containing `--color`
      rule.selectorText === ':root' &&
      rule.style &&
      Object.values(rule.style).some((decl) => decl.startsWith('--color'))
    )
    .map((rule) => Object.values(rule.style))
    .flat();

  // return removing duplicates
  return [...new Set(themeVariables)];
}
```

but the concept is more or less the same, you choose which one to use 🙃

Reply:
works with small changes